vpr/src/pack/cluster.cpp

```diff
@@ -2633,6 +2641,31 @@ static void echo_clusters(char* filename) {
         }
     }
 
+    fprintf(fp, "\nCLUSTER CONSTRAINTS:\n");
+    auto& floorplanning_ctx = g_vpr_ctx.mutable_floorplanning();
+
+    for (ClusterBlockId clb_id : cluster_ctx.clb_nlist.blocks()) {
+        std::vector<Region> reg = floorplanning_ctx.cluster_constraints[clb_id].get_partition_region();
+        if (reg.size() != 0) {
+            fprintf(fp, "\nRegions in Cluster %zu:\n", size_t(clb_id));
+            for (unsigned int i = 0; i < reg.size(); i++) {
+                vtr::Rect<int> rect = reg[i].get_region_rect();
+                int xmin = rect.xmin();
```

Comment:
make a print helper function for region, partitionregion, see if rect has a print function

Reply:
can also make a helper function for vtr::rect if there is not already one